db/compaction/compaction_iterator.cc

```diff
@@ -519,7 +520,8 @@ void CompactionIterator::NextFromInput() {
       // In the previous iteration we encountered a single delete that we could
       // not compact out.  We will keep this Put, but can drop it's data.
       // (See Optimization 3, below.)
-      if (ikey_.type != kTypeValue && ikey_.type != kTypeBlobIndex) {
+      if (ikey_.type != kTypeValue && ikey_.type != kTypeBlobIndex &&
+          ikey_.type != kTypeWideColumnEntity) {
```

Comment:
I think this line is part of the `SingleDelete` processing logic (not compaction filter), right?

Otherwise, yes, when an entity is split across column families, the compaction filter would get called for each part separately. (It might actually be a different compaction filter for each, since compaction filter is a CF option.)